sdk/internal/x/onending_processor.go

```diff
@@ -0,0 +1,14 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package x // import "go.opentelemetry.io/otel/sdk/internal/x"
+
+import "go.opentelemetry.io/otel/trace"
+
+// OnEndingSpanProcessor represents span processors that allow mutating spans
+// just before they are ended and made immutable.
+type OnEndingSpanProcessor interface {
+	// OnEnding is called while the span is finished, and while spans are still
+	// mutable. It is called synchronously and cannot block.
+	OnEnding(trace.Span)
```

Comment:
I think we should probably rely on `sdktrace.ReadWriteSpan` here, but I can't import the SDK, as it would be an import cycle.